File: frontend/src/services/teacher-ai.ts
import api from './api'

export interface TeacherAICourse {
  id: number
  name: string
  subject: string
  grade: string
  semester: string
}

export interface TeacherAIContextSummary {
  course_count: number
  lesson_plan_count: number
  resource_count: number
  student_count: number
  diagnosis_count: number
  classroom_count: number
  classroom_record_count: number
}

export interface TeacherAIContext {
  courses: TeacherAICourse[]
  summary: TeacherAIContextSummary
  sources: string[]
}

export interface TeacherAIHistoryMessage {
  role: 'user' | 'assistant'
  content: string
}

export interface TeacherAIReply {
  answer: string
  model: string
  course_id: number | null
  context_summary: TeacherAIContextSummary
  sources: string[]
}

export const teacherAIApi = {
  async getContext(courseId?: number): Promise<TeacherAIContext> {
    const response = await api.get('/teachers/ai-chat/context', {
      params: courseId ? { course_id: courseId } : undefined,
    })
    return response.data as TeacherAIContext
  },

  async sendMessage(
    message: string,
    history: TeacherAIHistoryMessage[],
    courseId?: number,
  ): Promise<TeacherAIReply> {
    const response = await api.post('/teachers/ai-chat/messages', {
      message,
      history,
      course_id: courseId ?? null,
    }, { timeout: 90000 })
    return response.data as TeacherAIReply
  },
}

File: backend/app/schemas/teacher_ai.py
"""Schemas for the teacher-workbench-only AI assistant."""

from typing import Literal, Optional

from pydantic import BaseModel, Field


class TeacherAIHistoryMessage(BaseModel):
    role: Literal["user", "assistant"]
    content: str = Field(..., min_length=1, max_length=8000)


class TeacherAIChatRequest(BaseModel):
    message: str = Field(..., min_length=1, max_length=4000)
    course_id: Optional[int] = Field(default=None, ge=1)
    history: list[TeacherAIHistoryMessage] = Field(default_factory=list, max_length=20)


class TeacherAIContextSummary(BaseModel):
    course_count: int = 0
    lesson_plan_count: int = 0
    resource_count: int = 0
    student_count: int = 0
    diagnosis_count: int = 0
    classroom_count: int = 0
    classroom_record_count: int = 0


class TeacherAIChatResponse(BaseModel):
    answer: str
    model: str
    course_id: Optional[int] = None
    context_summary: TeacherAIContextSummary
    sources: list[str] = Field(default_factory=list)
